### ISSUE TITLE
bug: Dismounting a player leaves visual hitbox in place, different from actual one

### ISSUE THREAD

**Issue by FunnyInternet** [rvbsm/fsit#31]
### 🔍 What happened?

1) If I get off of someone my game freezes for a moment and outputs error attached. 
2) If someone gets off of me (both if they do or I throw them off), a ghost version of them stays on my head, the actual location of the player is invisible and can hit me, I see only particles and such. This is fixed by relogging or leaving the area they are actually in (outputting the error).
3) If someone riding me disconnects, a ghost version of them will stay in place in air (which stays even if they come back)
![image](https://github.com/rvbsm/fsit/assets/57048295/82c11041-bd97-4319-8a14-d1fc2c888ffc)

The minecraft version exactly is 1.20.4
Fabric loader 0.15.6

### 🤔 What did you expect to happen?

For the feature to be usable without desyncing of player's visual hitbox location and their actual one

### 📦 Version

1.5.1

### ⛏ Minecraft Version

1.20.3

### 📃 Relevant log output

```bash
[18:16:19] [Render thread/ERROR]: Error executing task on Client
net.minecraft.class_148: Main thread packet handler
	at net.minecraft.class_2600.method_11072(class_2600.java:33) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_18859(class_1255.java:156) ~[client-intermediary.jar:?]
	at net.minecraft.class_4093.method_18859(class_4093.java:23) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_16075(class_1255.java:130) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_5383(class_1255.java:115) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1523(class_310.java:1283) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:888) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:265) ~[minecraft-1.20.4-client.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470) ~[fabric-loader-0.15.6.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) ~[fabric-loader-0.15.6.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) ~[fabric-loader-0.15.6.jar:?]
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:88) ~[NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:126) ~[NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:71) ~[NewLaunch.jar:?]
Caused by: java.lang.ClassCastException: class net.minecraft.class_746 cannot be cast to class net.minecraft.class_3222 (net.minecraft.class_746 and net.minecraft.class_3222 are in unnamed module of loader net.fabricmc.loader.impl.launch.knot.KnotClassLoader @64bf3bbf)
	at net.minecraft.class_1297.handler$bhl000$fsit$removePassenger$removeRider(class_1297.java:6382) ~[client-intermediary.jar:?]
	at net.minecraft.class_1297.method_5793(class_1297.java:2192) ~[client-intermediary.jar:?]
	at net.minecraft.class_1297.method_29239(class_1297.java:2153) ~[client-intermediary.jar:?]
	at net.minecraft.class_1657.method_29239(class_1657.java:1120) ~[client-intermediary.jar:?]
	at net.minecraft.class_1297.method_5848(class_1297.java:2158) ~[client-intermediary.jar:?]
	at net.minecraft.class_1309.method_5848(class_1309.java:2880) ~[client-intermediary.jar:?]
	at net.minecraft.class_1297.method_31745(class_1297.java:3658) ~[client-intermediary.jar:?]
	at net.minecraft.class_638.method_2945(class_638.java:338) ~[client-intermediary.jar:?]
	at net.minecraft.class_634.method_37472(class_634.java:621) ~[client-intermediary.jar:?]
	at it.unimi.dsi.fastutil.ints.IntArrayList.forEach(IntArrayList.java:702) ~[fastutil-8.5.12.jar:?]
	at it.unimi.dsi.fastutil.ints.IntIterable.forEach(IntIterable.java:146) ~[fastutil-8.5.12.jar:?]
	at it.unimi.dsi.fastutil.ints.AbstractIntCollection.forEach(AbstractIntCollection.java:153) ~[fastutil-8.5.12.jar:?]
	at net.minecraft.class_634.method_11095(class_634.java:620) ~[client-intermediary.jar:?]
	at net.minecraft.class_2716.method_11764(class_2716.java:31) ~[client-intermediary.jar:?]
	at net.minecraft.class_2716.method_11054(class_2716.java:9) ~[client-intermediary.jar:?]
	at net.minecraft.class_2600.method_11072(class_2600.java:24) ~[client-intermediary.jar:?]
	... 13 more
```

**Comment by rvbsm** [comment]
Interesting...

I'm currently in the process of refactoring where this issue _might_ be fixed, so I'll keep this issue open until the next release

**Comment by rvbsm** [comment]
Fixed in the v2.0.0, messed up with mixins. Thank you for the opening this issue!
